modules/programs/zimfw.nix

```diff
@@ -0,0 +1,157 @@
+    ## file. To do that with Nix, you'll need $ZIM_HOME to be writable.
+    ## `~/.cache/zim` is a good place for that. The problem is that zim also
+    ## looks for `zimfw.zsh` there, so we're going to tell it here to look for
+    ## the `zimfw.zsh` where we currently are.
+    postFixup = ''
+      echo "POSTINSTALL"
+      substituteInPlace $out/zimfw.zsh \
+        --replace "\''${ZIM_HOME}/zimfw.zsh" "$out/zimfw.zsh"
+    '';
+
+    meta = with lib; {
+      description =
+        "The Zsh configuration framework with blazing speed and modular extensions";
+      homepage = "https://zimfw.sh";
+      license = licenses.mit;
+      platforms = platforms.all;
+    };
+  };
+in {
+  meta.maintainers = [ maintainers.joedevivo ];
+
+  options = {
+    programs.zsh.zimfw = {
+      enable = mkEnableOption "Zim - ${zimPkg.meta.description}";
+
+      homeDir = mkOption {
+        default = "$HOME/.zim";
+        example = "$HOME/.cache/zim";
+        type = types.str;
+        description = ''
+          Working directory for zim. It stores downloaded plugins here.
+        '';
+      };
+
+      configFile = mkOption {
+        default = if cfg.dotDir != null then
+          "$HOME/${cfg.dotDir}/.zimrc"
+        else
+          "$HOME/.zimrc";
+        type = types.str;
+        description = ''
+          Location of zimrc file
+        '';
+      };
+
+      degit = mkOption {
+        default = true;
+        example = false;
+        type = types.bool;
+        description = ''
+          Use zim's degit tool for faster module installation.
+        '';
+      };
+
+      disableVersionCheck = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          disables 30 day version check
+        '';
+      };
+
+      zmodules = mkOption {
+        default = [ "environment" "git" "input" "termtitle" "utility" ];
+        example = [
+          "environment"
+          "git"
+          "input"
+          "termtitle"
+          "utility"
+          "exa"
+          "$PATH_TO_LOCAL_MODULE"
+          "duration-info"
+          "git-info"
+        ];
+        type = types.listOf types.str;
+        description = ''
+          List of zimfw modules. These are added to `.zimrc` verbatim,
+          so ENV vars will work for paths to local modules.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.zimfw.enable {
+    home.packages = [ zimPkg ];
+    programs.zsh.localVariables = {
+      ZIM_HOME = cfg.zimfw.homeDir;
+      ZIM_CONFIG_FILE = cfg.zimfw.configFile;
+    };
+    programs.zsh.initExtra = concatStringsSep "\n" ([
+      (optionalString (cfg.zimfw.degit) ''
+        zstyle ':zim:zmodule' use 'degit'
+      '')
+      (optionalString (cfg.zimfw.disableVersionCheck) ''
+        zstyle ':zim' disable-version-check yes
+      '')
+      ''
+        if [[ ! -e ${zimPkg}/zimfw.zsh ]]; then
+          echo "Zim is missing, this is a nix issue."
+        fi
```

Comment:
checking that this file exists should probably just be part of the test phase of the package